lib/TH/THVectorImpls.h

```diff
@@ -0,0 +1,602 @@
+            "beq         2f               @ \n\t"                       \
+            "vld1.32     {d16-d19}, [r1]! @ \n\t"                       \
+            "vld1.32     {d20-d23}, [r1]! @ \n\t"                       \
+            "vst1.32     {d0-d3}, [r2]!   @ \n\t"                       \
+            "vld1.32     {d0-d3}, [r0]!   @ \n\t"                       \
+            "vst1.32     {d4-d7}, [r2]!   @ \n\t"                       \
+            "vld1.32     {d4-d7}, [r0]!   @ \n\t"                       \
+            "b           1b               @ \n\t"                       \
+            "2:                           @ \n\t"                       \
+            "vst1.32     {d0-d3}, [r2]!   @ \n\t"                       \
+            "vst1.32     {d4-d7}, [r2]!   @ \n\t"                       \
+            "3:                           @ \n\t"                       \
+            "lsrs        r5, %3, #3       @ \n\t"                       \
+            "ands        r5, #1           @ \n\t"                       \
+            "beq         4f               @ \n\t"                       \
+            "vld1.32     {d16-d19}, [r1]! @ \n\t"                       \
+            "vld1.32     {d0-d3}, [r0]!   @ \n\t"                       \
+            "vmla.f32    q0, q8, q14      @ \n\t"                       \
+            "vmla.f32    q1, q9, q14      @ \n\t"                       \
+            "vst1.32     {d0-d3}, [r2]!   @ \n\t"                       \
+            "4:                           @ \n\t"                       \
+            "ands        r5, %3, #7       @ \n\t"                       \
+            "beq         6f               @ \n\t"                       \
+            "5:                           @ \n\t"                       \
+            "subs        r5, r5, #1       @ \n\t"                       \
+            "vld1.32     {d16[0]}, [r1]!  @ \n\t"                       \
+            "vld1.32     {d0[0]}, [r0]!   @ \n\t"                       \
+            "vmla.f32    d0, d16, d28     @ \n\t"                       \
+            "vst1.32     d0[0], [r2]!     @ \n\t"                       \
+            "bne         5b               @ \n\t"                       \
+            "6:                           @ "                           \
+            :                                                           \
+            :"r" (y),"r" (x), "r"(caddr),"r"(n)                         \
+            : "cc", "r0", "r1", "r2", "r5", "memory",                   \
+              "q0", "q1", "q2", "q3", "q14",                            \
+              "d0", "d1", "d2", "d3", "d4", "d5", "d6", "d7",           \
+              "d16", "d17", "d18", "d19", "d20", "d21", "d22", "d23", "d28", "d29" \
+            );                                                          \
+    }
+
+static inline void THDoubleVector_fill(double *x, const double c, const long n) {
```

Comment:
This is just code copied from `generic/THVector.c`. Adding NEON to dispatch tables would eliminate this duplication. If we don't choose to go this way, it would be better to generate them just for double from `generic/THVector.c` anyway.